lib/base_service.ts

```diff
@@ -23,7 +23,7 @@ import { AuthenticatorInterface, checkCredentials, readExternalSources } from '.
 import { stripTrailingSlash } from './helper';
 import { RequestWrapper } from './requestwrapper';
 
-export interface UserOptions {
+export interface UserOptions extends AxiosRequestConfig {
```

Comment:
Given that we export `UserOptions` from this package, doesn't making it extend `AxiosRequestConfig` expose the underlying transport mechanism of the core?  If so, I think that's not something we want to do.